File: Data Structures & Algorithms/validate-parentheses/submission-0.ts
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s: string): boolean {
        const map: Record<string, string> = {
            ")": "(",
            "}": "{",
            "]": "[",
        };

        const stack: string[] = [];
        const values = Object.values(map)

        for (const ch of s) {
            if (values.includes(ch)) stack.push(ch);
            else {
                if (stack.pop() !== map[ch]) {
                    return false;
                }
            }
        }

        return stack.length === 0;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        var stack = new Stack<int>();
        String[] operations = new string[] { "+", "-", "*", "/" };

        foreach (string token in tokens) {
            if (operations.Contains(token)) {
                int b = stack.Pop();
                int a = stack.Pop();

                int result = 0;

                if (token == "*")
                    result = a * b;
                else if (token == "+")
                    result = a + b;
                else if (token == "-")
                    result = a - b;
                else if (token == "/")
                    result = a / b;
                else
                    result = 0;

                stack.Push(result);
            }
            else
                stack.Push(int.Parse(token));
        }

        return stack.Pop();
    }
}
